internal/daemonset/daemonset.go

```diff
@@ -200,13 +204,17 @@ func (dc *daemonSetGenerator) SetDevicePluginAsDesired(ctx context.Context, ds *
 	}
 
 	return dc.constructDaemonSet(ctx, ds, mod, *mod.Spec.DevicePlugin, "device-plugin", "", standardLabels,
-		mod.Spec.Selector, containerVolumeMounts, dsVolumes, true)
+		nodeSelector, containerVolumeMounts, dsVolumes, true)
 }
 
 func (dc *daemonSetGenerator) IsDevicePluginDaemonSet(ds appsv1.DaemonSet) bool {
 	return ds.Labels[dc.kernelLabel] == ""
 }
 
+func (dc *daemonSetGenerator) GetDriverContainerNodeLabel(moduleName string) string {
```

Comment:
I don't think this needs to be a method.

Reply:
fixed